lib/components/SIndicator.vue

```diff
@@ -0,0 +1,89 @@
+<script setup lang="ts">
+import IconCheckCircleFill from '@iconify-icons/ph/check-circle-fill'
+import IconCircle from '@iconify-icons/ph/circle-bold'
+import IconCircleDashed from '@iconify-icons/ph/circle-dashed-bold'
+import IconCircleNotch from '@iconify-icons/ph/circle-notch-bold'
+import IconXCircle from '@iconify-icons/ph/x-circle-bold'
+import { computed } from 'vue'
+import SIcon from './SIcon.vue'
+
+export type Size = 'nano' | 'mini' | 'small' | 'medium' | 'large' | 'jumbo'
+export type State = 'pending' | 'ready' | 'queued' | 'running' | 'completed' | 'failed'
+export type Mode = 'colored' | 'monochrome'
+
+const props = withDefaults(defineProps<{
+  size?: Size
+  state: State
+  mode?: Mode
+}>(), {
+  size: 'medium',
+  mode: 'colored'
+})
+
+const classes = computed(() => [
+  props.size,
+  props.state,
+  props.mode
+])
+
+const stateIconDict = {
+  pending: IconCircle,
+  ready: IconCircle,
+  queued: IconCircleDashed,
+  running: IconCircleNotch,
+  completed: IconCheckCircleFill,
+  failed: IconXCircle
+}
+</script>
+
+<template>
+  <div class="SIndicator" :class="classes">
+    <SIcon class="icon" :icon="stateIconDict[state]" />
+  </div>
+</template>
+
+<style scoped lang="postcss">
+@keyframes indicator-blink {
+  0%   { opacity: 1; }
+  50%  { opacity: 0.5; }
+  100% { opacity: 1; }
+}
+
+@keyframes indicator-spin {
+  0%   { transform: rotate(0deg); }
+  100% { transform: rotate(360deg); }
+}
+
+.icon {
+  width: 100%;
+  height: 100%;
+}
+
+.SIndicator.nano   { width: 20px; height: 20px; }
+.SIndicator.mini   { width: 24px; height: 24px; }
+.SIndicator.small  { width: 28px; height: 28px; }
+.SIndicator.medium { width: 32px; height: 32px; }
+.SIndicator.large  { width: 40px; height: 40px; }
+.SIndicator.jumbo  { width: 48px; height: 48px; }
+
+.SIndicator.queued {
+  animation: indicator-blink 1s linear infinite;
+}
+
+.SIndicator.running {
+  animation: indicator-spin 1s linear infinite;
+}
```

Comment:
maybe increase their duration a bit, they look too fast currently.

Reply:
`1400ms cubic-bezier(0.45, 0.05, 0.55, 0.95) infinite` looks better (keep .running linear)

Reply:
Updated to 1.5s, since we use 0.25, 0.5, 0.75 scales for animations 👀  And added that bezier. Yeah it looks nicer 👍